zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessInstanceElementActivatingApplier.java

```diff
@@ -275,7 +275,7 @@ private void createEventScope(
       final var hasEvents = !eventSupplier.getEvents().isEmpty();
       if (hasEvents
           || flowElement instanceof ExecutableJobWorkerElement
-          || flowElement instanceof ExecutableCallActivity) {
+          || flowElement instanceof ExecutableActivity) {
```

Comment:
:wrench: We could restructure the condition to make the intention more clear.

If the element has events, create an event scope with the event ids. Otherwise, create an empty event scope. 

For example:

```
    if (flowElement instanceof final ExecutableCatchEventSupplier eventSupplier
        && !eventSupplier.getEvents().isEmpty()) {

      eventScopeInstanceState.createInstance(
          elementInstanceKey,
          eventSupplier.getInterruptingElementIds(),
          eventSupplier.getBoundaryElementIds());

    } else if (flowElement instanceof ExecutableJobWorkerElement
        || flowElement instanceof ExecutableActivity) {
      // job worker elements without events (e.g. message throw events)
      eventScopeInstanceState.createInstance(
          elementInstanceKey, Collections.emptySet(), Collections.emptySet());
    }
```

Reply:
Thanks @saig0 for the insightful suggestion and example! 🙏